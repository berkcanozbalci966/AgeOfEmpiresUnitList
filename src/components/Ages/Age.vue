<template>
  <div>
    <span> {{ ageInformation }} </span>
  </div>
</template>

<script>
export default {
  name: "Age",
  props: {
    ageInformation: {
      type: String,
      default: () => "",
    },
  },
};
</script>

<style lang="scss" scoped>
div {
  border: 1px solid black;
  padding: 4px;
  &:hover {
    cursor: pointer;
  }
}
</style>
