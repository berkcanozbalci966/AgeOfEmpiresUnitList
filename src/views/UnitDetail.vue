<template>
  <div>
    <UnitTable :getUnitInformations="getUnitInformations" />
  </div>
</template>

<script>
import { mapGetters } from "vuex";
import UnitTable from "@/components/UnitTable/Table";
export default {
  components: {
    UnitTable,
  },
  computed: {
    getUnitInformations() {
      const id = this.$route.params.id;
      return this.getUnitByID(id);
    },
    ...mapGetters("Units", ["getUnitByID"]),
  },
};
</script>

<style></style>
