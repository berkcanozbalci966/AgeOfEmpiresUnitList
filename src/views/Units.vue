<template>
  <div>
    <h1>Age List</h1>
    <Ages style="margin-bottom: 40px" />
    <h1>Cost Lists</h1>
    <Costs />
    <Table :unitsData="unitFormatedData" />
  </div>
</template>
<script>
import Ages from "@/components/Ages/Ages";
import Costs from "@/components/Costs/Costs";
import Table from "@/components/Table/Table";
import { mapState, mapGetters } from "vuex";
export default {
  components: {
    Ages,
    Costs,
    Table,
  },
  mounted() {
    console.log("mounted", this.unitState);
  },

  computed: {
    unitFormatedData() {
      return this.tableFormatter(this.getUnitsByAge(this.selectedAge));
    },
    ...mapGetters("Units", ["tableFormatter", "getUnitsByAge"]),
    ...mapState({
      unitState: (state) => state.Units.allUnitJson.units,
      selectedAge: (state) => state.Age.selectedAge,
    }),
  },
  methods: {},
};
</script>

<style lang="scss" scoped></style>
