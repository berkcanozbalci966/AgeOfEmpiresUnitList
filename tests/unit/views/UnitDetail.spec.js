import { mount } from "@vue/test-utils";
import Vuex from "vuex";
import Units from "../../../src/store/modules/units";
import Age from "../../../src/store/modules/age";
import UnitDetail from "../../../src/views/UnitDetail";

describe("Unit Detail", () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        Units,
        Age,
      },
    });
  });

  it("Render width route param Correctly", () => {
    const mockRoute = {
      params: {
        id: 3,
      },
    };
    const mockRouter = {
      push: jest.fn(),
    };
    // Mock Test
    const component = mount(UnitDetail, {
      global: {
        plugins: [store],
        mocks: {
          $route: mockRoute,
          $router: mockRouter,
        },
      },
    });
    expect(component.html()).toMatchSnapshot();
  });
});
