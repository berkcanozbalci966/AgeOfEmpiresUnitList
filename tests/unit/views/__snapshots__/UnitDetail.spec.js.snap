// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Detail Render width route param Correctly 1`] = `
<div>
  <table>
    <tr>
      <th scope="row">ID:</th>
      <td>3</td>
    </tr>
    <tr>
      <th scope="row">Name:</th>
      <td>Arbalest</td>
    </tr>
    <tr>
      <th scope="row">Description:</th>
      <td>Upgraded crossbowman</td>
    </tr>
    <tr>
      <th scope="row">Min.Required Age:</th>
      <td>Imperial</td>
    </tr>
    <tr>
      <th scope="row">Wood Cost:</th>
      <td>25</td>
    </tr>
    <tr>
      <th scope="row">Gold Cost:</th>
      <td>45</td>
    </tr>
    <!--v-if-->
    <tr>
      <th scope="row">Build Time:</th>
      <td>27</td>
    </tr>
    <tr>
      <th scope="row">Reload Time:</th>
      <td>2</td>
    </tr>
    <tr>
      <th scope="row">Hit Points:</th>
      <td>40</td>
    </tr>
    <tr>
      <th scope="row">Attack:</th>
      <td>6</td>
    </tr>
    <tr>
      <th scope="row">Accuracy:</th>
      <td>90%</td>
    </tr>
  </table>
</div>
`;
